Added !status

LW001 opened the pull request:
As described in https://trello.com/c/Do2gYoM4/.

Using the command works the following way:
`!status <ID/LINK> <NEW STATUS> <COMMENT>`
* ID/LINK works using the magical UVRegex(TM)
* NEW STATUS accepts the following values:
`review, re, under review, ur, u, planned, plan, pl, p, started, start, completed, complete, co, c, declined, decline, de, d`
* COMMENT accepts a comment to be added, also filtering out pipes.

UV endpoint: https://developer.uservoice.com/docs/api/reference/#_api_v1_forums_forum_id_suggestions_suggestion_id_respond_put

I actually ran a syntax check this time, are you proud yet

LW001 committed:
Added !status

LW001 committed:
Actually sending an ID now

Actually giving UV an ID and not a broom with the letters ID on it now

Dougley commented on Commands/Commands/uservoice.js:
```diff
@@ -207,6 +207,79 @@ commands.comment = {
+        let grabstatusidfromhere = statncontent.match(grabidregex)
+        let statusname = grabstatusidfromhere[1]
+        let statusid
+        if (statusname === 'review' || statusname === 're' || statusname === 'under review' || statusname === 'ur' || statusname === 'u') statusid = '1707873'
```
Use a regex

LW001 committed:
using regex

complain on im going to bed cya tomorrow

fathers day fuck yeah

SteamingMutt commented on Commands/Commands/uservoice.js:
```diff
@@ -219,14 +219,13 @@ commands.status = {
         s.shift()
         let statncontent = s.join(' ')
-        let grabstatusidfromhere = statncontent.match(grabidregex)
+        let grabstatusidfromhere = statncontent.toLowercase().match(grabidregex)
```
Use a case insensitive regex instead

Dougley commented on Commands/Commands/uservoice.js:
```diff
@@ -207,6 +207,78 @@ commands.comment = {
+        let idt = s[0]
+        s.shift()
+        let statncontent = s.join(' ')
+        let grabstatusidfromhere = statncontent.toLowercase().match(grabidregex)
```
Use a case insensitive regex instead

Dougley reviewed (changes requested):
Last nitpick before I'll merge this

LW001 committed:
Some changes from Trello

Numbers are now accepted
1. no status (Doesn't work yet)
2. under review
3. planned
4. started
5. completed
6. declined

Success message updated to fit Trello and to delete after 2 seconds as described in Trello
Giving out a status name instead of the ID now

LW001 committed:
Case insensitive regex

LW001 committed:
Able to remove status completely

Not sure whether this will work fully.

`null` value taken from https://developer.uservoice.com/docs/api/v2/reference/#/status_updates_1 which has `"old_status": null` in an example response. Important to note is also that the above link is from v2 so I'm not sure whether it'll work the same way in v1.

Possible command values to remove a status are `1, none, x`.

LW001 committed:
regex starts at 1, not 0

Should no longer analyze the ID for text

LW001 committed:
Fixed variable typo

LW001 committed:
reverted that because it wasnt the issue

LW001 committed:
Allowed not to add a comment

LW001 committed:
reverting reverting because sth broke

LW001 commented:
I fucked this one up quite a ton, cant edit it anymore because I deleted the repo, trying to get that fixed lul

LW001 commented:
Kinda won't be able to recover that repo, http://imgur.com/a/9CiLZ

Closing PR and reopening it later today, sorry for the inconvenience.
